hoomd/mpcd/RotatedParallelPlateGeometry.h

```diff
@@ -0,0 +1,165 @@
+// Copyright (c) 2009-2024 The Regents of the University of Michigan.
+// Part of HOOMD-blue, released under the BSD 3-Clause License.
+
+/*!
+ * \file mpcd/RotatedParallelPlateGeometry.h
+ * \brief Definition of the MPCD rotated slit channel geometry
+ */
+
+#ifndef MPCD_ROTATED_PARALLEL_PLATE_GEOMETRY_H_
+#define MPCD_ROTATED_PARALLEL_PLATE_GEOMETRY_H_
+
+#include "hoomd/BoxDim.h"
+#include "hoomd/HOOMDMath.h"
+
+#ifdef __HIPCC__
+#define HOSTDEVICE __host__ __device__ inline
+#else
+#define HOSTDEVICE inline __attribute__((always_inline))
+#include <string>
+#endif // __HIPCC__
+
+namespace hoomd
+    {
+namespace mpcd
+    {
+//! Rotated parallel plate (slit) geometry
+/*!
+ */
+class __attribute__((visibility("default"))) RotatedParallelPlateGeometry
+    {
+    public:
+    //! Constructor
+    /*!
+     * \param H Channel half-width
+     * \param angle Rotation angle (degrees)
+     * \param V Velocity of the wall
+     * \param no_slip Boundary condition at the wall (slip or no-slip)
+     */
+    HOSTDEVICE RotatedParallelPlateGeometry(Scalar H, Scalar angle, Scalar V, bool no_slip)
+        : m_H(H), m_angle(angle * M_PI / 180.), m_V(V), m_no_slip(no_slip)
+        {
+        }
+
+    //! Detect collision between the particle and the boundary
+    /*!
+     * \param pos Proposed particle position
+     * \param vel Proposed particle velocity
+     * \param dt Integration time remaining
+     *
+     * \returns True if a collision occurred, and false otherwise
+     *
+     * \post The particle position \a pos is moved to the point of reflection, the velocity \a vel
+     * is updated according to the appropriate bounce back rule, and the integration time \a dt is
+     * decreased to the amount of time remaining.
+     */
+    HOSTDEVICE bool detectCollision(Scalar3& pos, Scalar3& vel, Scalar& dt) const
+        {
+        // TODO: fill this in
+        const Scalar y1 = pos.x * tan(m_angle * M_PI / 180) + m_H;
+        const Scalar y2 = pos.x * tan(m_angle * M_PI / 180) - m_H;
+
+        const signed char sign = (char)((pos.y > y1) - (pos.y < y2));
+
+        if (sign == 0 || (vel.x * tan(m_angle * M_PI) - vel.y) == Scalar(0))
+            {
+            dt = Scalar(0);
+            return false;
+            }
+
+        if (sign == +1)
+            {
+            dt = (pos.y - y1) / (vel.x * tan(m_angle * M_PI) - vel.y);
+
+            pos.x -= vel.x * dt;
+            pos.y = y1;
+            pos.z -= vel.z * dt;
+            }
+
+        if (sign == -1)
+            {
+            dt = (y2 - pos.y) / (vel.x * tan(m_angle * M_PI) - vel.y);
+
+            pos.x -= vel.x * dt;
+            pos.y = y2;
+            pos.z -= vel.z * dt;
+            }
+
+        if (m_no_slip)
+            {
+            vel.x = -vel.x + Scalar(sign * 2) * m_V;
+            vel.z = -vel.z;
+            }
+        // both slip and no-slip have no penetration of the surface
+        vel.y = -vel.y;
```

Comment:
This reflection needs to be done a bit differently from before because we want to reflect the tangent (no-slip) and normal (always) components of the velocity. So, we need to decompose the x and y components of the velocity before reflection into tangent and normal. Then, the wall velocity will get added to the tangent velocity. Finally, we add the reflected tangent and normal components back together.

We can discuss the vector math together! The closest example I have is over in Yashraj & Kritika's repo:

https://github.com/YMWani/hoomd-blue/blob/62e845eaa8cab376d2bb7baa976fcc13e5e502e2/hoomd/mpcd/SphereGeometry.h#L116-L147

Note though that this is for a sphere, and the surface is moving in the normal (not tangent) direction.